src/librustc_typeck/astconv.rs

```diff
@@ -1276,11 +1278,26 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
 
         self.prohibit_generics(slice::from_ref(item_segment));
 
+        // Check if we have an enum variant here.
+        if let ty::Adt(adt_def, _) = ty.sty {
+            if adt_def.is_enum() {
+                if allow_type_alias_enum_variants(tcx, ty_hir, span) {
```

Comment:
Feature gates do not *enable* features (with exception of very rare cases), they gate them.
This should always return `Def::Varian`, but possibly with an accompanying feature error.

Reply:
It didn't return `Def::Variant` before, however. That's what I'm worried about. Will we be maintaining backwards compatibility then?